Arrangement_on_surface_2/include/CGAL/Arrangement_2/Arrangement_on_surface_2_impl.h

```diff
@@ -2739,14 +2739,22 @@ _insert_at_vertices(DHalfedge* he_to,
       he1->set_inner_ccb(ic1);
       he2->set_inner_ccb(ic1);
 
-      // Make all halfedges along ic2 to point to ic1.
-      DHalfedge* curr;
-
-      for (curr = he2->next(); curr != he1; curr = curr->next())
-        curr->set_inner_ccb(ic1);
+      if (m_sweep_mode)
+      {
+        CGAL_assertion(ic1->is_valid());
+        CGAL_assertion(ic2->is_valid());
```

Comment:
You can add a comment that the precondition are valid because both ic1 and ic2 are obtained using `inner_ccb()` and that this function is doing path compression.

Reply:
Indeed.